sql/core/src/main/java/org/apache/spark/sql/sources/v2/writer/DataSourceWriter.java

```diff
@@ -62,6 +62,16 @@
    */
   DataWriterFactory<Row> createWriterFactory();
 
+  /**
+   * Returns whether Spark should use the commit coordinator to ensure that only one attempt for
```

Comment:
This is actually not a guarantee, is it?

Reply:
Currently, the commit coordinator will only authorize one attempt and only authorize another if the authorized attempt fails, so it does ensure that only one attempt commits. Do you think the wording here needs to change?

Instead of documenting the behavior of the commit coordinator here, I'd rather point to its docs. Are those written, or is the coordinator an internal class?

Reply:
`only one` -> `at most one`? BTW I think we should not link to an internal class.

Reply:
I agree we shouldn't link to an internal class, but I don't think this is the place to document the built-in coordinator's behavior. Is there already a doc for that elsewhere that is public?